include/sys/vdev_impl.h

```diff
@@ -272,6 +272,8 @@ struct vdev {
 	    MAX_UBERBLOCK_SHIFT)
 #define	VDEV_UBERBLOCK_COUNT(vd)	\
 	(VDEV_UBERBLOCK_RING >> VDEV_UBERBLOCK_SHIFT(vd))
+#define	VDEV_UBERBLOCK_SLOT(vd, txg)	\
+	(txg % VDEV_UBERBLOCK_COUNT(vd))
```

Comment:
I see what you mean.   ZFS has lots and lots of uint64_t variables, including txg, but most operations on them seem at a quick glance to be assignment, comparison, shift, or add.
I'll think about it more.